src/sonic-yang-models/yang-models/sonic-bgp-neighbor.yang

```diff
@@ -48,7 +48,56 @@ module sonic-bgp-neighbor {
 
     container sonic-bgp-neighbor {
         container BGP_NEIGHBOR {
+            list BGP_NEIGHBOR_TEMPLATE_LIST {
+                description "This list is to support template based BGP neighbor configuration handled by bgpcfgd";
+                key "neighbor";
```

Comment:
Can we add example config in sample config_db.json

Reply:
I did in my original commit but Alex mentioned both BGP_NEIGHBOR based on frrcfgd and bgpcfgd cant co-exist and hence I removed it.

Reply:
My specific complaint was the organization of the json file having two keys. One would have overwritten the other and it would not have tested well. 

Given this is a unit test I think it is acceptable to have both forms of the config to validate that both forms are correctly parsed / recongnized by the model.